### DISCUSSION TITLE
Checkpoint system does not work with an explicit solver

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#20718] [Q&A General] [answered]
Trying to use `--recover` with an explicit calculation leads to the following error 

```
[2]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[2]PETSC ERROR: Object is in the wrong state
[2]PETSC ERROR: Matrix is missing diagonal entry 0
[2]PETSC ERROR: See https://petsc.org/release/faq/ for trouble shooting.
[2]PETSC ERROR: Petsc Release Version 3.16.5, unknown
 ```

I have been using the checkpoint system without issue for the same simulation but with an implicit time integrator.

Here is my executioner block
```
[Executioner]
  type = Transient
  end_time = 9e-5
  dt = 8e-10
  solve_type = LINEAR
  dtmin = 1e-20

  [TimeIntegrator]
    type = CentralDifference
    use_constant_mass = true
  []
[]
```

I can give more information as needed, but before that I want to check if this is an inherent limitation that I am unaware of. 

Thank you in advance

**Comment by GiudGiud** [comment, marked as answer]
Hello

recover and central difference time integration should work, it's tested by the test suite.

`use_constant_mass = true` might not be supported though
might also be the `solve_type = LINEAR` that is an issue. 

Could you please try to comment these out and report on your findings? (they are optimizations, the code should run without)

Guillaume

**Reply by TheGreatCid** [reply]
Thank you for your answer! 
It seems that the issue was with `use_constant_mass = true`
Without that option enabled I able to use the checkpoint system

**Reply by GiudGiud** [reply]
great thanks for reporting this. I ll submit an issue and we ll fix this at some point.
